<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>addEntry</class>
 <widget class="QDialog" name="addEntry">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>920</width>
    <height>568</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <widget class="QLabel" name="label_6">
   <property name="geometry">
    <rect>
     <x>90</x>
     <y>110</y>
     <width>81</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>New Concept</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_5">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>180</y>
     <width>49</width>
     <height>14</height>
    </rect>
   </property>
   <property name="text">
    <string>Tags</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_4">
   <property name="geometry">
    <rect>
     <x>350</x>
     <y>480</y>
     <width>101</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Need to solve again ?</string>
   </property>
  </widget>
  <widget class="QComboBox" name="sites_combobox">
   <property name="geometry">
    <rect>
     <x>110</x>
     <y>18</y>
     <width>141</width>
     <height>24</height>
    </rect>
   </property>
  </widget>
  <widget class="QDateEdit" name="attemptDate">
   <property name="geometry">
    <rect>
     <x>100</x>
     <y>48</y>
     <width>110</width>
     <height>24</height>
    </rect>
   </property>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>240</x>
     <y>48</y>
     <width>101</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Date of Completion</string>
   </property>
  </widget>
  <widget class="QWidget" name="layoutWidget">
   <property name="geometry">
    <rect>
     <x>180</x>
     <y>110</y>
     <width>48</width>
     <height>48</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QRadioButton" name="nc_y">
      <property name="text">
       <string>Yes</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QRadioButton" name="nc_no">
      <property name="text">
       <string>No</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>48</y>
     <width>49</width>
     <height>14</height>
    </rect>
   </property>
   <property name="text">
    <string>Date oF attempt</string>
   </property>
  </widget>
  <widget class="QLabel" name="url">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>88</y>
     <width>49</width>
     <height>14</height>
    </rect>
   </property>
   <property name="text">
    <string>URL</string>
   </property>
  </widget>
  <widget class="QDateEdit" name="completionDate">
   <property name="geometry">
    <rect>
     <x>350</x>
     <y>38</y>
     <width>110</width>
     <height>24</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton">
   <property name="geometry">
    <rect>
     <x>630</x>
     <y>500</y>
     <width>85</width>
     <height>27</height>
    </rect>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_2">
   <property name="geometry">
    <rect>
     <x>720</x>
     <y>500</y>
     <width>85</width>
     <height>27</height>
    </rect>
   </property>
   <property name="text">
    <string>Cancel</string>
   </property>
  </widget>
  <widget class="QLCDNumber" name="lcdNumber">
   <property name="geometry">
    <rect>
     <x>490</x>
     <y>10</y>
     <width>64</width>
     <height>23</height>
    </rect>
   </property>
   <property name="focusPolicy">
    <enum>Qt::StrongFocus</enum>
   </property>
   <property name="toolTip">
    <string comment="Number of problems Solved Until Now" extracomment="Number of problems Solved Until Now a"/>
   </property>
   <property name="autoFillBackground">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>18</y>
     <width>81</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Coding Site</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="url_ledit">
   <property name="geometry">
    <rect>
     <x>100</x>
     <y>78</y>
     <width>113</width>
     <height>24</height>
    </rect>
   </property>
  </widget>
  <widget class="QWidget" name="layoutWidget_3">
   <property name="geometry">
    <rect>
     <x>470</x>
     <y>480</y>
     <width>48</width>
     <height>48</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <widget class="QRadioButton" name="nsa_y">
      <property name="text">
       <string>Yes</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QRadioButton" name="nsa_c">
      <property name="text">
       <string>No</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QLabel" name="lproblem_name">
   <property name="geometry">
    <rect>
     <x>260</x>
     <y>20</y>
     <width>81</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Problem Name</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="problem_name_ledit">
   <property name="geometry">
    <rect>
     <x>350</x>
     <y>10</y>
     <width>113</width>
     <height>24</height>
    </rect>
   </property>
  </widget>
  <widget class="QWidget" name="layoutWidget">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>210</y>
     <width>161</width>
     <height>248</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="tags1">
    <item>
     <widget class="QCheckBox" name="chk_Ad_hoc">
      <property name="text">
       <string>Adhoc</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QCheckBox" name="chk__2sat">
      <property name="text">
       <string>2sat</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QCheckBox" name="chk_bfs">
      <property name="text">
       <string>bfs</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QCheckBox" name="chk_binarySearch">
      <property name="text">
       <string>binarysearch</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QCheckBox" name="chk_bipartite_graph">
      <property name="text">
       <string>Bipartite_grap</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QCheckBox" name="chk_bit_manipulation">
      <property name="text">
       <string>Bit Manipulation</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QCheckBox" name="chk_bitmasks">
      <property name="text">
       <string>Bit Maska</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QCheckBox" name="chk_brute_force">
      <property name="text">
       <string>Brute Force</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QCheckBox" name="chk_combinatorics">
      <property name="text">
       <string>Combinatorics</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QCheckBox" name="chk_constructive_algorithms">
      <property name="text">
       <string>Constructive algorithmss</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="layoutWidget">
   <property name="geometry">
    <rect>
     <x>330</x>
     <y>210</y>
     <width>151</width>
     <height>248</height>
    </rect>
   </property>
   <layout class="QFormLayout" name="formLayout_2">
    <item row="0" column="0">
     <widget class="QCheckBox" name="chk_games">
      <property name="text">
       <string>Games</string>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QCheckBox" name="chk_gcd">
      <property name="text">
       <string>GCD</string>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QCheckBox" name="chk_geometry">
      <property name="text">
       <string>Geometry</string>
      </property>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="QCheckBox" name="chk_graphs">
      <property name="text">
       <string>Graphs</string>
      </property>
     </widget>
    </item>
    <item row="4" column="0">
     <widget class="QCheckBox" name="chk_greedy">
      <property name="text">
       <string>Greedy</string>
      </property>
     </widget>
    </item>
    <item row="5" column="0">
     <widget class="QCheckBox" name="chk_hash_map">
      <property name="text">
       <string>Hash Map</string>
      </property>
     </widget>
    </item>
    <item row="6" column="0">
     <widget class="QCheckBox" name="chk_hashing">
      <property name="text">
       <string>Hashing</string>
      </property>
     </widget>
    </item>
    <item row="7" column="0">
     <widget class="QCheckBox" name="chk_heap">
      <property name="text">
       <string>Heap</string>
      </property>
     </widget>
    </item>
    <item row="8" column="0">
     <widget class="QCheckBox" name="chk_implementation">
      <property name="text">
       <string>Implementation</string>
      </property>
     </widget>
    </item>
    <item row="9" column="0">
     <widget class="QCheckBox" name="chk_matrices">
      <property name="text">
       <string>Matrices</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="layoutWidget">
   <property name="geometry">
    <rect>
     <x>170</x>
     <y>210</y>
     <width>151</width>
     <height>248</height>
    </rect>
   </property>
   <layout class="QFormLayout" name="formLayout">
    <property name="sizeConstraint">
     <enum>QLayout::SetFixedSize</enum>
    </property>
    <item row="0" column="0">
     <widget class="QCheckBox" name="chk_datastructures">
      <property name="text">
       <string>Data Structure</string>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QCheckBox" name="chk_dfs">
      <property name="text">
       <string>DFS</string>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QCheckBox" name="chk_disjoint_set">
      <property name="text">
       <string>Disjoint set</string>
      </property>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="QCheckBox" name="chk_divideandconquer">
      <property name="text">
       <string>Divide and conq</string>
      </property>
     </widget>
    </item>
    <item row="4" column="0">
     <widget class="QCheckBox" name="chk_dsu">
      <property name="text">
       <string>dsu</string>
      </property>
     </widget>
    </item>
    <item row="5" column="0">
     <widget class="QCheckBox" name="chk_dynamic_programming">
      <property name="text">
       <string>Dynamic Programmin</string>
      </property>
     </widget>
    </item>
    <item row="6" column="0">
     <widget class="QCheckBox" name="chk_factorization">
      <property name="text">
       <string>Factorization</string>
      </property>
     </widget>
    </item>
    <item row="7" column="0">
     <widget class="QCheckBox" name="chk_fenwicktree">
      <property name="text">
       <string>Fenwick Tree</string>
      </property>
     </widget>
    </item>
    <item row="8" column="0">
     <widget class="QCheckBox" name="chk_fft">
      <property name="text">
       <string>fft</string>
      </property>
     </widget>
    </item>
    <item row="9" column="0">
     <widget class="QCheckBox" name="chk_flows">
      <property name="text">
       <string>Flows</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="layoutWidget">
   <property name="geometry">
    <rect>
     <x>650</x>
     <y>210</y>
     <width>151</width>
     <height>248</height>
    </rect>
   </property>
   <layout class="QFormLayout" name="formLayout_4">
    <item row="0" column="0">
     <widget class="QCheckBox" name="chk_sieve">
      <property name="text">
       <string>Sieve</string>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QCheckBox" name="chk_simulation">
      <property name="text">
       <string>Simulation</string>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QCheckBox" name="chk_sorting">
      <property name="text">
       <string>Sorting</string>
      </property>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="QCheckBox" name="chk_spanning_tree">
      <property name="text">
       <string>Spanning Tree</string>
      </property>
     </widget>
    </item>
    <item row="4" column="0">
     <widget class="QCheckBox" name="chk_sqrt_decomposition">
      <property name="text">
       <string>Sqrt Decomposition</string>
      </property>
     </widget>
    </item>
    <item row="5" column="0">
     <widget class="QCheckBox" name="chk_stack">
      <property name="text">
       <string>Stack</string>
      </property>
     </widget>
    </item>
    <item row="6" column="0">
     <widget class="QCheckBox" name="chk_string_suffix_structures">
      <property name="text">
       <string>String Suffix Structures</string>
      </property>
     </widget>
    </item>
    <item row="7" column="0">
     <widget class="QCheckBox" name="chk_suffix_arrays">
      <property name="text">
       <string>Suffix Arrays</string>
      </property>
     </widget>
    </item>
    <item row="8" column="0">
     <widget class="QCheckBox" name="chk_ternarysearch">
      <property name="text">
       <string>Ternary Search</string>
      </property>
     </widget>
    </item>
    <item row="9" column="0">
     <widget class="QCheckBox" name="chk_trees">
      <property name="text">
       <string>Trees</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="layoutWidget">
   <property name="geometry">
    <rect>
     <x>490</x>
     <y>210</y>
     <width>151</width>
     <height>248</height>
    </rect>
   </property>
   <layout class="QFormLayout" name="formLayout_3">
    <item row="0" column="0">
     <widget class="QCheckBox" name="chk_matrix_exponentiation">
      <property name="text">
       <string>Matrix Exponentiation</string>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QCheckBox" name="chk_meet_in_the_middle">
      <property name="text">
       <string>Meet in the middle</string>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QCheckBox" name="chk_memoization">
      <property name="text">
       <string>Memoization</string>
      </property>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="QCheckBox" name="chk_modular_arithmetic">
      <property name="text">
       <string>Modular Arithmetic</string>
      </property>
     </widget>
    </item>
    <item row="4" column="0">
     <widget class="QCheckBox" name="chk_numbertheory">
      <property name="text">
       <string>Number Theory</string>
      </property>
     </widget>
    </item>
    <item row="5" column="0">
     <widget class="QCheckBox" name="chk_priority_queue">
      <property name="text">
       <string>Priority Queue</string>
      </property>
     </widget>
    </item>
    <item row="6" column="0">
     <widget class="QCheckBox" name="chk_probabilty">
      <property name="text">
       <string>Probabilty</string>
      </property>
     </widget>
    </item>
    <item row="7" column="0">
     <widget class="QCheckBox" name="chk_recursion">
      <property name="text">
       <string>Recursion</string>
      </property>
     </widget>
    </item>
    <item row="8" column="0">
     <widget class="QCheckBox" name="chk_segment_trees">
      <property name="text">
       <string>Segment Trees</string>
      </property>
     </widget>
    </item>
    <item row="9" column="0">
     <widget class="QCheckBox" name="chk_shortestpath">
      <property name="text">
       <string>Shortest Path</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="">
   <property name="geometry">
    <rect>
     <x>810</x>
     <y>210</y>
     <width>91</width>
     <height>48</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <item>
     <widget class="QCheckBox" name="chk_trie">
      <property name="text">
       <string>Trie</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QCheckBox" name="chk_twopointers">
      <property name="text">
       <string>Two Pointer</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <zorder>layoutWidget</zorder>
  <zorder>layoutWidget</zorder>
  <zorder>layoutWidget</zorder>
  <zorder>layoutWidget</zorder>
  <zorder>layoutWidget</zorder>
  <zorder>label_6</zorder>
  <zorder>label_5</zorder>
  <zorder>label_4</zorder>
  <zorder>sites_combobox</zorder>
  <zorder>attemptDate</zorder>
  <zorder>label_3</zorder>
  <zorder>layoutWidget</zorder>
  <zorder>label_2</zorder>
  <zorder>url</zorder>
  <zorder>completionDate</zorder>
  <zorder>pushButton</zorder>
  <zorder>pushButton_2</zorder>
  <zorder>lcdNumber</zorder>
  <zorder>label</zorder>
  <zorder>url_ledit</zorder>
  <zorder>layoutWidget_3</zorder>
  <zorder>lproblem_name</zorder>
  <zorder>problem_name_ledit</zorder>
  <zorder>chk_trie</zorder>
  <zorder>chk_twopointers</zorder>
 </widget>
 <tabstops>
  <tabstop>sites_combobox</tabstop>
  <tabstop>problem_name_ledit</tabstop>
  <tabstop>attemptDate</tabstop>
  <tabstop>completionDate</tabstop>
  <tabstop>url_ledit</tabstop>
  <tabstop>nc_y</tabstop>
  <tabstop>nc_no</tabstop>
  <tabstop>chk_Ad_hoc</tabstop>
  <tabstop>chk__2sat</tabstop>
  <tabstop>chk_bfs</tabstop>
  <tabstop>chk_binarySearch</tabstop>
  <tabstop>chk_bipartite_graph</tabstop>
  <tabstop>chk_bit_manipulation</tabstop>
  <tabstop>pushButton</tabstop>
  <tabstop>chk_bitmasks</tabstop>
  <tabstop>chk_brute_force</tabstop>
  <tabstop>chk_combinatorics</tabstop>
  <tabstop>chk_constructive_algorithms</tabstop>
  <tabstop>chk_datastructures</tabstop>
  <tabstop>chk_dfs</tabstop>
  <tabstop>chk_disjoint_set</tabstop>
  <tabstop>chk_divideandconquer</tabstop>
  <tabstop>chk_dsu</tabstop>
  <tabstop>chk_dynamic_programming</tabstop>
  <tabstop>chk_factorization</tabstop>
  <tabstop>chk_fenwicktree</tabstop>
  <tabstop>chk_fft</tabstop>
  <tabstop>chk_flows</tabstop>
  <tabstop>chk_games</tabstop>
  <tabstop>chk_gcd</tabstop>
  <tabstop>chk_graphs</tabstop>
  <tabstop>chk_geometry</tabstop>
  <tabstop>chk_hashing</tabstop>
  <tabstop>chk_heap</tabstop>
  <tabstop>chk_matrices</tabstop>
  <tabstop>chk_matrix_exponentiation</tabstop>
  <tabstop>chk_meet_in_the_middle</tabstop>
  <tabstop>chk_memoization</tabstop>
  <tabstop>chk_modular_arithmetic</tabstop>
  <tabstop>chk_numbertheory</tabstop>
  <tabstop>chk_priority_queue</tabstop>
  <tabstop>chk_probabilty</tabstop>
  <tabstop>chk_recursion</tabstop>
  <tabstop>chk_segment_trees</tabstop>
  <tabstop>chk_shortestpath</tabstop>
  <tabstop>chk_sieve</tabstop>
  <tabstop>chk_simulation</tabstop>
  <tabstop>chk_sorting</tabstop>
  <tabstop>chk_spanning_tree</tabstop>
  <tabstop>chk_sqrt_decomposition</tabstop>
  <tabstop>chk_stack</tabstop>
  <tabstop>chk_string_suffix_structures</tabstop>
  <tabstop>chk_suffix_arrays</tabstop>
  <tabstop>chk_ternarysearch</tabstop>
  <tabstop>chk_trees</tabstop>
  <tabstop>lcdNumber</tabstop>
  <tabstop>pushButton_2</tabstop>
  <tabstop>nsa_y</tabstop>
  <tabstop>nsa_c</tabstop>
  <tabstop>chk_implementation</tabstop>
  <tabstop>chk_hash_map</tabstop>
  <tabstop>chk_greedy</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
